test/lib/utils/helper.ts

```diff
@@ -24,6 +24,18 @@ export const approveToken = async (
     token: Address,
     amount = MAX_UINT256, // approve max by default
 ): Promise<boolean> => {
+    const allowance = await client.readContract({
```

Comment:
Break this out into a separate `hasAllowance` type helper.

Reply:
I was annoyed by the fact that I had to comment out token approvals for USDT because it fails if you try to approve it twice, so I added that, but I agree it makes sense to do it on a separate function.
I'll refactor over here 👍